src/openrct2/core/FixedPoint.hpp

```diff
@@ -21,7 +21,23 @@ using fixed32_2dp = int32_t;
 using fixed64_1dp = int64_t;
 
 // Construct a fixed point number. For example, to create the value 3.65 you
-// would write FIXED_2DP(3,65)
-#define FIXED_XDP(x, whole, fraction) ((whole) * (10 * (x)) + (fraction))
-#define FIXED_1DP(whole, fraction) FIXED_XDP(1, whole, fraction)
-#define FIXED_2DP(whole, fraction) FIXED_XDP(10, whole, fraction)
+// would write MakeFixed2dp(3, 65)
+template<typename T, uint8_t numDecimals>
+static constexpr T MakeFixedXdp(T whole, uint8_t fraction)
```

Comment:
I wonder if we should turn this into an int8_t, so we can statically assert the `fraction >= 0 && fraction <= 100` as is implicitly expected.

Reply:
Can we not do that with the uint8_t already here? E.g. `fraction <= 100`?

Reply:
My concern was also that we might get overflows going undetected. With an int8_t, those would become negative and therefore likely detectable.

Maybe this is overly cautious, though. @ZehMatt, any thoughts?

Reply:
Keep in mind that overflow/underflow of signed integers in C++ is UB.

Reply:
I wasn’t able to add `static_assert` to these functions anyway. Not quite sure why.